### PR TITLE
Dev amine

### PR DESCRIPTION
**Pull request by amine-zzr**
#2 issue is handled

### CODE REVIEW

File: mandatory.c [hunk 1]
```diff
@@ -30,3 +30,32 @@ int _putchar(int c)
...
+ * Return: On success, the number of bytes printed is returned.
+ */
+int print_num(int n, int base)
+{
+	int counter = 0;
+	char *numbers = "0123456789";
+
+	if (n < 0)
+	{
+		write(1, "-", 1);
```
Comment (rakati) [inline mandatory.c]: write function can return -1 if an error occurred we need to handle that

File: mandatory.c [hunk 2]
```diff
@@ -30,3 +30,32 @@ int _putchar(int c)
...
+		return (print_num(-n, base) + 1);
```
Comment (rakati) [inline mandatory.c:55]: the MIN_INT will overflow when multiplying by -1.

File: mandatory.c [hunk 3]
```diff
@@ -30,3 +30,32 @@ int _putchar(int c)
...
+	}
+	else if (n < base)
+		return (_putchar(numbers[n]));
+	else if (n > base)
+	{
+		counter = print_num(n / base, base);
+		return (counter + (print_num(n % base, base)));
+	}
+	else
+		return (-1);
```
Comment (rakati) [inline mandatory.c]: There is no need for an else just `return (-1)` is enough.
the same as above, an if statement with a return may not necessarily need an else following it.

File: test/test_main.c [hunk 4]
```diff
@@ -27,5 +27,13 @@ int main(void)
 
 	printf("--------------------\n");
 
+	/* test decimal numbers */
+	_printf("Length:[%d, %i]\n", len1, len1);
+	printf("Length:[%d, %i]\n", len2, len2);
+	_printf("Negative:[%d]\n", -762534);
+	printf("Negative:[%d]\n", -762534);
```
Comment (rakati) [inline test/test_main.c]: add test 0, test MAX_INT, MIN_INT.

File: mandatory.c (hunk 3 above)
Comment (amine-zzr) [reply mandatory.c]: i didn't quite understand your point about the above else if statements

**Review by rakati — APPROVED** [review]

**Pull request merged**